sample_apps/run_example.sh

```diff
@@ -647,11 +1149,64 @@ function run_steps() {
 # Do initial setup of the sample app test-case
 # ###########################################################################
 function setup() {
-    run_steps "show_env" "provision_app_service_files"
+    local app_name="$1"
+    case "${app_name}" in
+        "simple_app")
+            setup_simple_app
+            ;;
+
+        "simple_app_under_oe")
+            setup_simple_app_for_oe ""
+            ;;
+
+        *)
+            echo "${Me}: Unknown sample app name '${app_name}'."
+            return
+            ;;
+    esac
+}
+
+# ###########################################################################
+# Do initial setup of the sample app test-case
+# As the sub-steps array has nested fn-calls, run a collection of steps
+# to avoid re-running sub-fns nested under author_policy()
+# ###########################################################################
+function setup_simple_app() {
+    run_steps "show_env" "get_measurement_of_trusted_app"
+    author_policy
+    run_steps "build_simple_server" "provision_app_service_files"
+}
+
+# ###########################################################################
+# For OE-app, we have two ways to manage policy generation.
+# Default is manual step by running a collection of utility programs.
+#
+# Optionally, user can hand-edit policy.json file and stick-in the
+# generated measurement. This is also automated under 'generate_policy' step.
+# So, user can invoke this script as:
+#   $ run_example.sh simple_app_under_oe setup generate_policy_for_OE
+#
+# If the 2nd arg is provided, manually execute that sub-step's function.
+# ###########################################################################
+function setup_simple_app_for_oe() {
+    local setup_arg="$1"
+    run_steps "show_env" "get_measurement_of_trusted_app"
+    if [ "${setup_arg}" = "" ]; then
+        author_policy_for_OE
+    else
+        ${setup_arg}
+    fi
+    run_steps "build_simple_server" "provision_app_service_files"
+}
+
+# ###########################################################################
+function setup_with_generate_policy() {
```

Comment:
This name needs a re-think. This is implementing the alternative Step 8: Use Policy Generator.

What better name can we come up to self-document the actual implementation of editing this json file etc.

Reply:
"Policy Generator" is the name of the tool

Reply:
Changed fn's name as we discussed to `setup_with_auto_policy_generation_for_OE()`